### ISSUE TITLE
Exporting member containers as read-only fields

### ISSUE THREAD

**Issue by mehatesgithub** [ThePhD/sol2#273]
When binding usertype one can export fields of a container type (vector, map and so on). But if one wants to export this container as a read-only property, the default method with `sol::readonly` does not work.

```cpp
struct Foo
{
    std::list<Bar> bar;
};

lua.new_usertype<Foo>(
    "foo",
    "bar", &Foo::bar  // this one works
//  "bar", sol::readonly(&Foo::bar)  // this one does not work
);
```

It compiles, but any attempt to use this field in lua generates following error:
`sol: 'self' argument is not the proper type (pass 'self' as first argument with ':' or call on proper type)`

**Comment by ThePhD** [comment]
My bad. This is a bug in the implementation.

**Comment by ThePhD** [comment]
All fixed now and backported in 2.15.1: https://github.com/ThePhD/sol2/releases/tag/v2.15.1
